WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java

```diff
@@ -212,23 +226,45 @@ private static class GapMarkerViewHolder extends RecyclerView.ViewHolder {
 
     @Override
     public int getItemViewType(int position) {
-        if (position == 0 && hasSiteHeader()) {
-            // first item is a ReaderSiteHeaderView
-            return VIEW_TYPE_SITE_HEADER;
-        } else if (position == 0 && hasTagHeader()) {
-            // first item is a ReaderTagHeaderView
-            return VIEW_TYPE_TAG_HEADER;
-        } else if (position == mGapMarkerPosition) {
+        // first item logic
```

Comment:
Just want to share my .2$ on this method. It seems to me that it covers multiple conditions, which could lead to confusion (and errors) if more conditions are added in the future. Maybe not in this PR, but we could refactor it by splitting the logic into multiple helper methods to improve readability and maintainability, such as 
adding methods like getFirstItemViewType() and getSecondItemViewType().

Reply:
That's a good point but I think doing so might be more confusing since the code falls through in case it doesn't match the conditions for Announcement / Headers, so even the first and second items could have a Post or Gap View Type.

That means that a `getFirstItemViewType` function would also need to return `null` or an invalid type to continue with the logic inside `getItemViewType`, which in my opinion would be harder to read and reason about.

That said, I found an improvement opportunity to make the code more readable and maintainable by separating the logic by type instead of position, similar to what is done for the Gap Marker, which makes it way cleaner IMO. See commit: https://github.com/wordpress-mobile/WordPress-Android/pull/20872/commits/f7c9246d4f809d905e385eef4b2c08108217374e